### PR TITLE
Update Colvars library to version 2016-12-22

### PR DESCRIPTION
**Pull request by giacomofiorin**
This PR includes significant code cleanup and several fixes (e.g. walls + extended Lagrangian).
There is also a new harmonicWalls bias to apply confining boundaries with time-dependent force constant & integration.

The README of the USER-COLVARS package and doc file of fix colvars have also been updated (last updated in 2015) to reflect changes in the code and in the LAMMPS development model, where contributors can now submit code directly via pull requests.

### CODE REVIEW

File: doc/src/fix_colvars.txt [hunk 1]
```diff
@@ -10,7 +10,7 @@ fix colvars command :h3
 
 [Syntax:]
 
-fix ID group-ID colvars configfile keyword values ... :pre
+fix ID group-ID Colvars configfile keyword values ... :pre
```
Comment (akohlmey) [inline doc/src/fix_colvars.txt]: this looks like an search-replace error. "colvars" should be lower case.

**Comment by akohlmey** [comment]
FYI, no merges until next week.

File: doc/src/fix_colvars.txt (hunk 1 above)
Comment (giacomofiorin) [inline doc/src/fix_colvars.txt]: No, it's carelessly semi-intentional: I got mixed up with the ID field, which I set to "Colvars" (upper case) in the library README.

**Comment by giacomofiorin** [comment]
No hurry for merges anyway

File: doc/src/fix_colvars.txt (hunk 1 above)
Comment (giacomofiorin) [inline doc/src/fix_colvars.txt]: Typo is now fixed.

**Pull request merged**